redfish-core/lib/log_services.hpp

```diff
@@ -6397,74 +6430,45 @@ inline void getRedfishUriByDbusObjPath(
                 return;
             }
 
-            while (uriIdPos != std::string::npos)
+            for (const auto& ancestorIface : ancestorsIfaces)
             {
-                std::string parentRedfishUri = redfishUri.substr(0,
-                                                                 uriIdPos - 1);
-                RedfishUriListType::const_iterator parentRedfishUriIt =
-                    std::find_if(redfishUriList.begin(), redfishUriList.end(),
-                                 [parentRedfishUri](const auto& ele) {
-                    return parentRedfishUri == ele.second;
-                });
-
-                if (parentRedfishUriIt == redfishUriList.end())
+                bool foundAncestor = false;
+                for (const auto& obj : ancestors)
                 {
-                    BMCWEB_LOG_ERROR(
-                        "Failed to fill Links:OriginOfCondition because unable to get parent Redfish URI [{}] DBus interface for the identified Redfish URI: {} of the given DBus object path: {} ",
-                        parentRedfishUri, redfishUri, dbusObjPath.str);
-                    messages::internalError(asyncResp->res);
-                    return;
-                }
-
-                std::string parentObj;
-                for (const auto& obj : subtree)
-                {
-                    if (dbusObjPath.str.find(
-                            sdbusplus::message::object_path(obj.first)
-                                .filename()) == std::string::npos)
-                    {
-                        // The object is not found in the isolated
-                        // hardware object path
-                        continue;
-                    }
-
                     for (const auto& service : obj.second)
                     {
                         for (const auto& interface : service.second)
                         {
-                            if (interface == parentRedfishUriIt->first)
+                            if (interface == ancestorIface.first)
                             {
-                                parentObj = obj.first;
+                                foundAncestor = true;
+                                redfishUri.replace(
+                                    ancestorIface.second, uriIdPattern.length(),
+                                    getIsolatedHwItemId(
+                                        sdbusplus::message::object_path(
+                                            obj.first)));
                                 break;
                             }
                         }
-                        if (!parentObj.empty())
+                        if (foundAncestor)
                         {
                             break;
                         }
                     }
-                    if (!parentObj.empty())
+                    if (foundAncestor)
                     {
                         break;
                     }
                 }
```

Comment:
L6433-L6461 is to find a matching one.
So, can it be done by making a sub-function?

Reply:
Sure. Will address this after the cherry-picking is completed, as there are few more commits letft from, https://github.com/ibm-openbmc/bmcweb/pull/542/commits and they are looking at getting everything into 1110 at the earliest